src/swap/Ord_Marketplace.sol

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {BTCUtils} from "@bob-collective/bitcoin-spv/BTCUtils.sol";
+import {BitcoinTx} from "../bridge/BitcoinTx.sol";
+import {IRelay} from "../bridge/IRelay.sol";
+import {BridgeState} from "../bridge/BridgeState.sol";
+
+using SafeERC20 for IERC20;
+
+contract Ord_Marketplace {
+    using BitcoinTx for BridgeState.Storage;
+
+    mapping(uint256 => OrdinalSellOrder) public ordinalSellOrders;
+    mapping(uint256 => AcceptedOrdinalSellOrder) public acceptedOrdinalSellOrders;
+
+    uint256 nextOrdinalId;
+    uint256 public constant REQUEST_EXPIRATION_SECONDS = 6 hours;
+
+    BridgeState.Storage internal relay;
+
+    constructor(IRelay _relay) {
+        relay.relay = _relay;
+        relay.txProofDifficultyFactor = 1; // will make this an arg later on
+    }
+
+    function setRelay(IRelay _relay) internal {
+        relay.relay = _relay;
+    }
+
+    event placeOrdinalSellOrderEvent(uint256 indexed orderId, bytes ordinalID, address sellToken, uint256 sellAmount);
+    event acceptOrdinalSellOrderEvent(
+        uint256 indexed id, uint256 indexed acceptId, BitcoinAddress bitcoinAddress, uint256 ercAmount, address ercToken
+    );
+    event proofOrdinalSellOrderEvent(uint256 id);
+    event withdrawOrdinalSellOrderEvent(uint256 id);
+    event cancelAcceptedOrdinalSellOrderEvent(uint256 id);
+
+    struct OrdinalSellOrder {
+        bytes ordinalID;
+        address sellToken;
+        uint256 sellAmount;
+        BitcoinTx.UTXO utxo;
+        address requester;
+    }
+
+    struct AcceptedOrdinalSellOrder {
+        uint256 orderId;
+        BitcoinAddress bitcoinAddress;
+        address ercToken;
+        uint256 ercAmount;
+        address requester;
+        address acceptor;
+        uint256 acceptTime;
+    }
+
+    struct BitcoinAddress {
+        bytes scriptPubKey;
+    }
+
+    struct OrdinalId {
+        bytes ordinalID;
```

Comment:
We could split this into `txid` and `index`, but since we're only displaying it in the UI this is probably fine

Reply:
added a todo for now